core/gui/src/app/dashboard/service/user/ai-assistant/ai-assistant.service.ts

```diff
@@ -27,4 +27,49 @@ export class AiAssistantService {
         return "NoAiAssistant";
       });
   }
+
+  public getTypeAnnotations(code: string, lineNumber: number, allcode: string): Promise<string> {
```

Comment:
don't return `Promise<string>`. for frontend, use `Subscription<>`. Also string is not meaningful at all. you should declare the type such as 
```
export type TypeAnnotationResponse = {
    type: string
}
```
and use `Subscription<TypeAnnotationResponse>`.